system-addon/lib/DownloadsManager.jsm

```diff
@@ -0,0 +1,183 @@
+ChromeUtils.import("resource://gre/modules/Services.jsm");
+Cu.importGlobalProperties(["URL"]);
+
+const {actionTypes: at} = ChromeUtils.import("resource://activity-stream/common/Actions.jsm", {});
+
+ChromeUtils.defineModuleGetter(this, "DownloadsViewUI",
+  "resource:///modules/DownloadsViewUI.jsm");
+ChromeUtils.defineModuleGetter(this, "DownloadsCommon",
+  "resource:///modules/DownloadsCommon.jsm");
+
+const DOWNLOAD_CHANGED_DELAY_TIME = 1000; // time in ms to delay timer for downloads changed events
+
+class DownloadElement extends DownloadsViewUI.DownloadElementShell {
+  constructor(download, browser) {
+    super();
+    this._download = download;
+    this.element = browser;
+    this.element._shell = this;
+  }
+
+  get download() {
+    return this._download;
+  }
+
+  get fileType() {
+    if (!this.download.target.path) {
+      return null;
+    }
+    let items = this.download.target.path.split(".");
+    return items[items.length - 1].toUpperCase();
+  }
+
+  downloadsCmd_copyLocation() {
+    let clipboard = Cc["@mozilla.org/widget/clipboardhelper;1"].getService(Ci.nsIClipboardHelper);
+    clipboard.copyString(this.download.source.url);
+  }
+
+  downloadsCmd_openReferrer() {
+    this.element.openNewTabWith(this.download.source.referrer);
```

Comment:
Looks like this opens in a background tab. Maybe just pass in true as the second argument for shifted?